src/client/debugger/jupyter/debuggingManager.ts

```diff
@@ -241,19 +241,25 @@ export class DebuggingManager implements IExtensionSingleActivationService, IDeb
                 switch (mode) {
                     case KernelDebugMode.Everything:
                         await this.startDebugging(editor.document);
-                        this.updateToolbar(true);
+                        if (this.isDebugging(editor.document)) {
+                            this.updateToolbar(true);
+                        }
                         break;
                     case KernelDebugMode.Cell:
                         if (cell) {
                             await this.startDebuggingCell(editor.document, KernelDebugMode.Cell, cell);
-                            this.updateToolbar(true);
+                            if (this.isDebugging(editor.document)) {
```

Comment:
Under what circustances will this `if` condition be `false`, i don't think this can ever be `false`, its aleways `true`, it should be true, else the previous await should throw an error.
Having this `if` makes it look as though the previous code can run to completion without errors & not start the debugger (which i think is impossible)

Reply:
I guess that's true, it just feels safer.
I'll remove them